platformio/debug/config/blackmagic.MD

```diff
@@ -0,0 +1,28 @@
+# Blackmagic Probe improvements
+
+I needed SRST under reset feature, this command has to be issued before scan, but this is currently impossible with current variables.
+
+By changing the gdb init a bit this or even target power is possible
+
+See: "Embedded Debugging with the Black Magic Probe", pp. 28
+
+```
+upload_protocol = blackmagic
+upload_port = /dev/ttyBmpGdb
+debug_tool = blackmagic
+debug_port = ${this.upload_port} connect_srst
+monitor_port = /dev/ttyBmpTarg
+monitor_speed = 115200
+
+```
+
+debug_port = port [monitor_cmd]
+
+debug_port can be /dev/ttyACM0 (/dev/ttyBmpGdb if you use the udev rules) or a Windows COM port, I use `${this.upload_port}` to use the same.
```

Comment:
Small nitpick, /dev/ttyACM0 is unreliable in name - the kernel gives us no guarantees that it didn't bring BMP up as some other name instead, hence the project udev rules. Second note, if a user wishes to connect to a specific BMP by serial number, they can update their udev rules and use /dev/ttyBmpGdb<serial>. The udev rules and serial number stuff is stable through releases